pinot-core/src/main/java/org/apache/pinot/core/transport/InstanceRequestHandler.java

```diff
@@ -154,10 +200,23 @@ public void onSuccess(@Nullable byte[] responseBytes) {
 
       @Override
       public void onFailure(Throwable t) {
+        if (_enableQueryCancellation) {
+          String queryId = queryRequest.getQueryId();
+          if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Remove track of running query: {} on failure", queryId);
+          }
+          _queryFuturesById.remove(queryId);
+        }
         // Send exception response.
-        LOGGER.error("Exception while processing instance request", t);
+        Exception ex = new Exception(t);
+        if (t instanceof CancellationException) {
+          LOGGER.info("Query: {} got cancelled", queryRequest.getQueryId());
+          ex = (Exception) t;
+        } else {
+          LOGGER.error("Exception while processing instance request", t);
+        }
```

Comment:
Not introduced in this PR, but suggest avoid wrapping exceptions
```suggestion
        Exception e;
        if (t instanceof Exception) {
          e = (Exception) t;
          if (e instanceof CancellationException) {
            LOGGER.info("Query: {} got cancelled", queryRequest.getQueryId());
          } else {
            LOGGER.error("Exception while processing instance request", e);
          }
        } else {
          LOGGER.error("Error while processing instance request", t);
          e = new Exception(t);
        }
```

Reply:
yeah, should have been so.